proxyfilters/op.go

```diff
@@ -20,20 +20,30 @@ var RecordOp = filters.FilterFunc(func(cs *filters.ConnectionState, req *http.Re
 	resp, nextCtx, err := next(cs, req)
 	if err != nil {
 		op.FailIf(err)
-		// Dumping stack trace is useful but in this case, it would create tons
-		// of noises as the filters are called recursively.
-		if e, ok := err.(errors.Error); ok {
-			// Before we do this, we should check if this is a DNS error, if so we don't
-			// want to log it, since these log.Error's get sent up to stack driver and become
-			// super noisy!
-			var dnsError *net.DNSError
-			if !e2.As(e.RootCause(), &dnsError) {
-				log.Error(e.RootCause())
-			}
-		} else {
-			log.Error(err)
-		}
+		logFilterError(err)
 	}
 	op.End()
 	return resp, nextCtx, err
 })
+
+func logFilterError(err error) {
+	var (
+		opErr  *net.OpError
+		dnsErr *net.DNSError
+	)
+	switch {
+	case stderrors.As(err, &opErr) && opErr.Timeout():
+		// Network timeouts are out of our control and create noise in StackDriver.
+		log.Debugf("%s timeout error: %v", opErr.Op, err)
+		return
+	case stderrors.As(err, &dnsErr):
```

Comment:
Previously, the `errors.As` check was on `err.RootCause`, after asserting that `err` implements the `getlantern.errors.Error` interface.  Since `getlantern.errors.Error` [implements error unwrapping](https://github.com/getlantern/errors/blob/1b7978a3b243cdf65d1c77a762538a356c3d8593/errors.go#L322-L326), we can simplify by just calling `errors.As` on the original `err`.